### ISSUE TITLE
Scenarios newly created from Scenario Editor start with incorrect temperature in relation to selected climate type

### ISSUE THREAD

**Issue by SpartanFrederic104** [OpenRCT2/OpenRCT2#20196]
### Operating System

Windows 10, 64-bit

### OpenRCT2 build

OpenRCT2, v0.4.5 (76ca840) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Describe the issue

When you create a new scenario in the Scenario Editor and then begin a new game in that scenario, most of the time it will start with a temperature of 55°F (13°C) and will quite often go into heavy rain near the beginning, even if the scenario in question has its climate set to Hot and dry. Previously, any newly created scenario would begin at 62°F (17°C) if its selected climate was Warm or Hot and dry, but now that is no longer the case. Every newly created scenario now starts with a temperature of 55°F (13°C), which is typical of the Cool and wet climate.
![Plagenockite Deadlands 2023-05-12 19-09-46](https://github.com/OpenRCT2/OpenRCT2/assets/61885939/7941c5d5-1668-463b-b8bc-6a4715c46618)
In the above screenshot I have created a new custom scenario. I selected the Warm climate while setting it up, and in older versions of OpenRCT2, as well as (quite possibly) RCT2 vanilla and RCT Classic, any new scenario created with Warm or Hot and dry climate would be expected to begin at 62°F (17°C). Yet, it starts at 55°F (13°C).

**EDIT (2023-05-13):** Can also confirm that this happens with the Cold climate setting, which incorrectly starts at 55°F (13°C) instead of 48°F (9°C) like it used to in older versions.

**EDIT2 (2023-06-23):** The game always selects Cool and wet as the default scenario climate when you create a new scenario.

### Steps to reproduce

1. Go to Scenario Editor
2. Create a new scenario and set its climate to Warm, Hot and dry, or Cold
3. Start up the scenario
4. Notice that the temperature now starts at 55°F (13°C) instead of 62°F (17°C) despite the scenario not using the Cool and wet climate setting

**Comment by duncanspumpkin** [comment]
Do you think we should always reset the climate if we change it? (i.e. even if you modify it in game save using the scenario options window via the cheats)
Or only reset it if you change it and are in the editor?

Eitherway its easy enough to fix:
1. If first option ClimateSetAction.cpp L46 should reset the climate (if it changed).
2. Or second option L1111 of EditorScenarioOptions.cpp needs a callback to reset the climate.

**Comment by Ajeets6** [comment]
Hi,I would like to take this issue and can you elaborate on the solution.

**Comment by duncanspumpkin** [comment]
> Hi,I would like to take this issue and can you elaborate on the solution.

Should be pretty simple just call climate reset from the game action execute function.

**Comment by c-pg-vega** [comment]
Has this issues been resolved yet? Id like to take it on!

**Comment by SpartanFrederic104** [comment]
@c-pg-vega Go right ahead!

**Comment by duncanspumpkin** [comment]
> @SpartanFrederic104 Hi so I am trying to fix this issue but I am a little lost on how to go about it - there is just alot of code at once and I am not sure where to begin here - from what I am reading from the previous comments made. I think I just need to find the game execution function (not sure where that it) and call reset climate from there

You want to go to the GameActions folder and find ClimateSetAction.cpp. there will be two main functions query and execute. You need to add a climate reset to the execute.

**Comment by tanzimchowdh** [comment]
@duncanspumpkin I am also trying to fix this issue. Based on the previous comments I think I've made the necessary changes.
![Screenshot 2023-12-02 231332](https://github.com/OpenRCT2/OpenRCT2/assets/55206233/7046616f-fa48-4b04-adfd-adc2fef58e70)
Will this suffice or should I call reset on gclimate instead and delete the assignment to ClimateType( _climate). Could you also assign this issue to me?

**Comment by duncanspumpkin** [comment]
> @duncanspumpkin I am also trying to fix this issue. Based on the previous comments I think I've made the necessary changes. ![Screenshot 2023-12-02 231332](https://private-user-images.githubusercontent.com/55206233/287496746-7046616f-fa48-4b04-adfd-adc2fef58e70.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3MTYyMTEsIm5iZiI6MTcwMTcxNTkxMSwicGF0aCI6Ii81NTIwNjIzMy8yODc0OTY3NDYtNzA0NjYxNmYtZmE0OC00YjA0LWFkZmQtYWRjMmZlZjU4ZTcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA0VDE4NTE1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0ZDVkNGQxMWFjMzlkODE3MGE5ZjVjYzczYmJlYjM0YzcyYzM5ZWEzZTM1YTkzYTEwMmQ5NTA0YmExMzc5NWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JHaRB1BAcD82DpJJh-sPSjCPsvXQUz6ejOI45h5SKCw) Will this suffice or should I call reset on gclimate instead and delete the assignment to ClimateType( _climate). Could you also assign this issue to me?

You'll have to look into what the function is doing and come to your own conclusion i can't remember other than it needs a reset.

**Comment by tanzimchowdh** [comment]
@duncanspumpkin , On line 76 of GameState.cpp, in the InitAll function, I see ClimateReset(ClimateType::CoolAndWet). Setting this enum value for Climate Type may be causing this issue to occur in the first place. As far as what ClimateReset() does, it simply takes the ClimateType parameter and initalizes gClimateCurrent elements based on that, e.g. Weather, Temperature, etc. Since I'm new to the project, I'm not sure what Execute() in this file actually does. I've tried to trace its references but nothing fruitful.

**Comment by tanzimchowdh** [comment]
> > @duncanspumpkin , On line 76 of GameState.cpp, in the InitAll function, I see ClimateReset(ClimateType::CoolAndWet). Setting this enum value for Climate Type may be causing this issue to occur in the first place. As far as what ClimateReset() does, it simply takes the ClimateType parameter and initalizes gClimateCurrent elements based on that, e.g. Weather, Temperature, etc. Since I'm new to the project, I'm not sure what Execute() in this file actually does. I've tried to trace its references but nothing fruitful.
> 
> Sounds like you can get rid of the `gClimate = ClimateType( _climate );` then and only do `ClimateReset(_climate);`. What is causing this issue is the fact `gClimateCurrent` isn't being changed when the climate is changed. You can test this by creating scenarios in the scenario editor and adjusting the climate (ScenarioOptions -> ParkOptions -> Climate drop down). Or you can adjust it with the ingame console.

After creating a scenario and selecting the climate type, I haven't been able to reproduce this issue. I'm also not completely sure on what the process is to see my code changes in effect. I've built in vscode and then run from developer prompt with and without my code changes; the issue still does not occur

**Comment by tanzimchowdh** [comment]
@duncanspumpkin, I retract my previous comment. I was able to recreate the bug. I also noticed that doing a double save of the same climate causes it to actually change so that it does not start incorrectly with the previously set climate. The code change I have made makes it so that a double save is not necessary through the scenario editor and the scenario starts with the correct climate after each save. I will make a PR.
